### PR TITLE
Fixes save error when model inherit from both TimeStampModel and StatusModel.

### PR DESCRIPTION
**Pull request by hramezani**
## Problem

Fixed https://github.com/jazzband/django-model-utils/issues/465

## Solution

Explain the solution that has been implemented, and what has been changed.

## Commandments

- [x] Write PEP8 compliant code.
- [x] Cover it with tests.
- [ ] Update `CHANGES.rst` file to describe the changes, and quote according issue with `GH-<issue_number>`.
- [ ] Pay attention to backward compatibility, or if it breaks it, explain why.
- [ ] Update documentation (if relevant).

### CODE REVIEW

**Comment by hramezani** [comment]
@arseniy-panfilov Could you please take a look at this patch?

File: model_utils/models.py [hunk 1]
```diff
@@ -32,10 +32,8 @@ def save(self, *args, **kwargs):
         a parameter to the update field argument.
         """
         update_fields = kwargs.get('update_fields', None)
-        if update_fields is not None:
-            update_fields = set(update_fields)
-            if update_fields:
-                kwargs['update_fields'] = update_fields.union({'modified'})
+        if update_fields and 'modified' not in update_fields:
+            kwargs['update_fields'] = list(kwargs['update_fields']) + ['modified']
```
Comment (charettes) [inline model_utils/models.py:36]: I think the previous approach was fine.

```suggestion
        if update_fields:
            kwargs['update_fields'] = set(update_fields).union({'modified'})
```

File: model_utils/models.py [hunk 2]
```diff
@@ -74,12 +72,14 @@ def save(self, *args, **kwargs):
...
         if (
-            'update_fields' in kwargs
-            and 'status' in kwargs['update_fields']
-            and 'status_changed' not in kwargs['update_fields']
+            update_fields
+            and 'status' in update_fields
+            and 'status_changed' not in update_fields
         ):
-            kwargs['update_fields'] += ['status_changed']
+            kwargs['update_fields'] = list(kwargs['update_fields']) + ['status_changed']
```
Comment (charettes) [inline model_utils/models.py:81]: I think we should keep using `set` as above.

```suggestion
            kwargs['update_fields'] = set(update_fields).union({'status_changed'})
```

Comment (bugraaydin) [reply model_utils/models.py:81]:
```
        update_fields = kwargs.get('update_fields', None)
        if update_fields is not None:
            update_fields = set(update_fields)
            if (
                update_fields
                and 'status' in update_fields
            )
        ):
            kwargs['update_fields'] = update_fields.union({'status_changed'})
```

changing the code here to above would do the trick i suppose

File: model_utils/models.py (hunk 1 above)
Comment (hramezani) [reply model_utils/models.py]: Thanks @charettes  I changed it to set approach

**Review by charettes — APPROVED** [review]
Thanks for the changes @hramezani 🙇

**Comment by hramezani** [comment]
Thanks for the review and suggestion @charettes.
Should we prepare a bug fix release?

**Comment by charettes** [comment]
@hramezani yeah I think it's worth doing. I can tag and release once this is merged if you can adjust the `CHANGES` with these adjustments.

**Comment by bugraaydin** [comment]
> Thanks for the review and suggestion @charettes.
> Should we prepare a bug fix release?

sorry about causing the headache here again, thanks for the quick fix

**Comment by hramezani** [comment]
@charettes I added an entry to `CHANGES`

**Pull request merged**